packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -261,8 +261,30 @@ class _PosixUtils extends OperatingSystemUtils {
   @override
   String get pathVarSeparator => ':';
 
+  HostPlatform _hostPlatform;
+
   @override
-  HostPlatform hostPlatform = HostPlatform.linux_x64;
+  HostPlatform get hostPlatform {
+    if (_hostPlatform == null) {
+      final RunResult hostPlatformCheck =
+          _processUtils.runSync(<String>['uname', '-m']);
+      // On x64 stdout is "uname -m: x86_64"
+      // On arm64 stdout is "uname -m: aarch64, arm64_v8a"
+      if (hostPlatformCheck.exitCode != 0) {
+        _logger.printTrace(
```

Comment:
This should at least be `_logger.printTrace()`, and maybe even `throwsToolExit()`. What do you think @jonahwilliams ?

Reply:
I guess my concern with throwToolExit is that we would break some weird system where this was happily working by chance and now fails. I would say definitely make this an printError though.

Reply:
Makes sense. `_logger.printError` sounds good to me.

Reply:
I'll fix from '_logger.printTrace' to '_logger.printError'.

Reply:
I'm done.